commercetools.Sdk/IntegrationTests/commercetools.Sdk.IntegrationTests/Messages/MessagesIntegrationTests.cs

```diff
@@ -0,0 +1,145 @@
+using commercetools.Sdk.Domain.Messages;
+using commercetools.Sdk.Domain.Messages.Customers;
+using commercetools.Sdk.Domain.Messages.Reviews;
+using commercetools.Sdk.Domain.Query;
+using commercetools.Sdk.Domain.Reviews;
+using Xunit;
+using static commercetools.Sdk.IntegrationTests.Customers.CustomersFixture;
+using static commercetools.Sdk.IntegrationTests.Reviews.ReviewsFixture;
+
+namespace commercetools.Sdk.IntegrationTests.Messages
+{
+    /// <summary>
+    /// Assume that messages enabled in danger zone of current project
+    /// </summary>
+    [Collection("Integration Tests")]
+    public class MessagesIntegrationTests
+    {
+        private readonly IClient client;
+
+        public MessagesIntegrationTests(ServiceProviderFixture serviceProviderFixture)
+        {
+            this.client = serviceProviderFixture.GetService<IClient>();
+        }
+
+        [Fact]
+        public async Task GetMessageById()
+        {
+            //lets create customer first to ensure message created
+            await WithCustomer(client, async customer =>
+            {
+                var queryCommand = new QueryCommand<Message>();
+                queryCommand.SetLimit(1);
+                var returnedSet = await client.ExecuteAsync(queryCommand);
+                Assert.Single(returnedSet.Results);
+                var messageId = returnedSet.Results[0].Id;
+
+                var retrievedMessage = await client
+                    .ExecuteAsync(new GetByIdCommand<Message>(messageId));
+                Assert.NotNull(retrievedMessage);
+            });
+        }
+
+        [Fact]
+        public async Task QuerySpecificTypeOfMessages()
+        {
+            //lets create customer first to ensure message created
+            await WithCustomer(client, async customer =>
+            {
+                var queryCommand = new QueryCommand<CustomerCreatedMessage>();
+                queryCommand.Where(message => message.Type == "CustomerCreated");
+                queryCommand.SetLimit(1);
+                queryCommand.SetSort(new List<Sort<CustomerCreatedMessage>>()
+                {
+                    new Sort<CustomerCreatedMessage>(message => message.CreatedAt, SortDirection.Descending)
+                });
+                var returnedSet = await client.ExecuteAsync(queryCommand);
+                Assert.Single(returnedSet.Results);
+                var customerCreatedMessage = returnedSet.Results[0];
+                Assert.NotNull(customerCreatedMessage);
+                Assert.NotNull(customerCreatedMessage.Customer);
+                Assert.Equal(customer.Id, customerCreatedMessage.Customer.Id);
+            });
+        }
+
+        [Fact]
+        public async Task QueryMultipleMessagesTypesOfSpecificResourceType()
+        {
+            //lets create customer first to ensure message created
+            await WithCustomer(client, DefaultCustomerDraftWithAddress, async customer =>
+            {
```

Comment:
Same here. Query for the messages belonging to the customer directly in an assert eventually style.

Reply:
I did the same here